mq/esworker_pubsub.py

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# Copyright (c) 2017 Mozilla Corporation
+
+import json
+import sys
+import socket
+import time
+from configlib import getConfig, OptionParser
+from datetime import datetime
+from mozdef_util.utilities.toUTC import toUTC
+from mozdef_util.utilities.logger import logger, initLogger
+from mozdef_util.elasticsearch_client import (
+    ElasticsearchClient,
+    ElasticsearchBadServer,
+    ElasticsearchInvalidIndex,
+    ElasticsearchException,
+)
+from google.cloud import pubsub
+from lib.plugins import sendEventToPlugins, registerPlugins
+
+# running under uwsgi?
+try:
+    import uwsgi
+
+    hasUWSGI = True
+except ImportError as e:
+    hasUWSGI = False
+
+
+class PubSubtaskConsumer(object):
+    def __init__(self, esConnection, options):
+        self.esConnection = esConnection
+        self.pluginList = registerPlugins()
+        self.options = options
+        self.scopes = ["https://www.googleapis.com/auth/cloud-platform", "https://www.googleapis.com/auth/pubsub "]
+        self.credentials_file = options.credentials_file
+
+    def run(self):
+        # XXX: fetch from the config file
+        subscriber = pubsub.SubscriberClient.from_service_account_file(self.options.credentials_file)
```

Comment:
Should we have an example credentials file added to mozdef source, so we know what the format of that file is?

Reply:
We do most of the time, I would pass on it here. The reason being - the whole file is downloaded from the GCP console and used as it is.

Reply:
We could have a placeholder telling people to download the file with credentials.